### PR TITLE
[RNMobile] Fixed position of `FloatingToolbar` on the bottom of the screen

### PR DESCRIPTION
**Pull request by jbinda**
## Description

This PR changes behaviour of `FloatingToolbar` to align with guideline from [this issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1414#issuecomment-603972811). It will be positioning absolutely at the bottom of the screen.

Please also refer to:
[Related gutenberg-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2118)
[Related gutenberg-mobile issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1414#issuecomment-603972811)

Testing branch on WP iOS and WP Android:
WordPress-iOS:  [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/13927#issuecomment-615245960)
WordPress-Android: [here](https://github.com/wordpress-mobile/WordPress-Android/pull/11692#issuecomment-615247791)

## How has this been tested?

1. Add below code in `initial-html`

<details>

```
/**
 * @format
 * @flow
 */

export default `
<!-- wp:media-text {"mediaId":1,"mediaType":"image"} -->
<div class="wp-block-media-text is-stacked-on-mobile"><figure class="wp-block-media-text__media"><img src="https://cldup.com/cXyG__fTLN.jpg" class="wp-image-1"/></figure><div class="wp-block-media-text__content"><!-- wp:paragraph {"className":"has-large-font-size"} -->
<p class="has-large-font-size"></p>
<!-- /wp:paragraph --></div></div>
<!-- /wp:media-text -->

<!-- wp:image -->
<figure class="wp-block-image"><img src="https://cldup.com/cXyG__fTLN.jpg" alt=""/></figure>
<!-- /wp:image -->

<!-- wp:group -->
<div class="wp-block-group"><div class="wp-block-group__inner-container"></div></div>
<!-- /wp:group -->

<!-- wp:group -->
<div class="wp-block-group"><div class="wp-block-group__inner-container"><!-- wp:image {"id":1,"sizeSlug":"large"} -->
<figure class="wp-block-image size-large"><img src="https://cldup.com/cXyG__fTLN.jpg" alt="" class="wp-image-1"/></figure>
<!-- /wp:image -->

<!-- wp:heading -->
<h2>Test heading</h2>
<!-- /wp:heading --></div></div>
<!-- /wp:group -->
`;
```

</details>

2. Feel free to add more nested structure
3. Check if each time you select nested block FloatingToolbar is visible on bottom of the screen

## Screenshots 

**LTR vs RTL mode**

## Types of changes

Refactor - change the FloatingToolbar placement and behaviour

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/block-list/block.native.js
Comment (jbinda) [inline packages/block-editor/src/components/block-list/block.native.js:126]: with slot-fill architecture on `FloatingToolbar` we can also move all the content to FloatingToolbar file and use it here as below:

`<FloatingToolbar clientId={ clientId } />`

Does it make sense or we would like to keep current PR version?

File: packages/block-editor/src/components/block-list/index.native.js
Comment (jbinda) [inline packages/block-editor/src/components/block-list/index.native.js:125]: Rendering slot only for RootList should eliminate render FloatingToolbar for each nested level

File: packages/block-editor/src/components/block-mobile-floating-toolbar/styles.native.scss
Comment (jbinda) [inline packages/block-editor/src/components/block-mobile-floating-toolbar/styles.native.scss:16]: This is temporary to set proper space of 8px between bottom toolbar and FloatingToolbar. Will refactor soon

Comment (jbinda) [reply packages/block-editor/src/components/block-mobile-floating-toolbar/styles.native.scss]: fixed!

File: packages/block-editor/src/components/block-list/block.native.js
Comment (jbinda) [reply packages/block-editor/src/components/block-list/block.native.js]: @dratwas I refactor to above according to our conversation

**Comment by cameronvoell** [comment]
Found a new crash on this PR tapping the toolbar button to split a Group block into separate blocks.
Steps: 
1. Add a group block with one or more blocks
2. Tap Arrow to select entire Group block
3. Tap toolbar button to split group block into separate blocks
4. Crash, see screen shot below

image

**Comment by jbinda** [comment]
thanks for reporting ! Will check that today

**Comment by jbinda** [comment]
I also noticed that it crash after try to remove any block. Already fixed that. 

The cause was that after ungroup (or delete) block breadcrumbs can not read block icon.

I have also add suport for RTL according to [this comment](https://github.com/wordpress-mobile/gutenberg-mobile/issues/2133#issuecomment-610998527)

File: packages/block-editor/src/components/block-list/breadcrumb.native.js
Comment (dratwas) [inline packages/block-editor/src/components/block-list/breadcrumb.native.js:105]: Why you added it here instead of using it directly? You could even move this logic to the `SubdirectorSVG`. WDYT?

File: packages/block-editor/src/components/block-list/index.native.js
Comment (dratwas) [reply packages/block-editor/src/components/block-list/index.native.js:127]: Hmm, maybe we could move it somewhere else. i.e `packages/edit-post/src/components/layout/index.native.js`. Do you think it is possible?

File: packages/block-editor/src/components/block-mobile-floating-toolbar/index.native.js
Comment (dratwas) [inline packages/block-editor/src/components/block-mobile-floating-toolbar/index.native.js:1]: This is a mobile-specific component. Please move it to the `packages/components/mobile/`

File: packages/block-editor/src/components/block-mobile-floating-toolbar/nav-up-icon.js
Comment (dratwas) [inline packages/block-editor/src/components/block-mobile-floating-toolbar/nav-up-icon.js:6]: Since it is only used for mobile I suggest to move the import of  I18nManager here and remove it from the floating toolbar

File: packages/block-editor/src/components/block-mobile-floating-toolbar/styles.native.scss
Comment (dratwas) [inline packages/block-editor/src/components/block-mobile-floating-toolbar/styles.native.scss:32]:
```suggestion
	height: $mobile-floating-toolbar-height;
```

Comment (jbinda) [reply packages/block-editor/src/components/block-mobile-floating-toolbar/styles.native.scss:32]: done.

File: packages/block-editor/src/components/block-list/index.native.js
Comment (jbinda) [reply packages/block-editor/src/components/block-list/index.native.js:127]: As discussed on call - moved it to layout component

File: packages/block-editor/src/components/block-list/breadcrumb.native.js
Comment (jbinda) [reply packages/block-editor/src/components/block-list/breadcrumb.native.js:105]: I refactored to use it directly and keep login here according to what we discussed

**Comment by pinarol** [comment]
@jbinda Could you test with Keyboard open? The FloatingToolbar will probably cover the caret on a Paragraph block.

On Android, the autoscroll is system managed, we don't do a custom thing for it so it can be a problem. 

On iOS, we are doing it with custom implementation but it can still be a problem because we haven't tried setting the extra scroll amount to a dynamic value so far. We now need to add the height of the FloatingToolbar to the extra scroll amount when it is visible.

**Comment by jbinda** [comment]
@pinarol I have do some testing and below is the results:

When you trying to select multiline text based block you can noticed that after scroll FloatingToolbar will cover the caret in text field - see below GIFs





On iOS we can solve that by adding FloatingToolbar height when it's visible to `inputAccessoryViewHeight`prop in `KeyboardAwareFlatList` as @pinarol suggested. On Android we currently do not have solution for that.


Alternative solution works for both is putting the `FLoatingToolbar` to the top of the screen. It solves mentioned problems. See below GIFs

   



@iamthomasbishop what do you think ?

**Comment by jbinda** [comment]
Another think is that I have noticed different behaviour depending on where we render `FloatingToolbar.SLot`

Initially I have render`<FloatingToolbar.SLot />` in BlockListBlock like this `{isRootList && <FloatingToolbar.Slot /> }` but then we will need to adjust position on iOS after keyboard opens. Without that it will be covered (also the position of additional toolbar is wrong)



With latest discussion with @dratwas we conclude that we can move `<FloatingToolbar.Slot />` to layout component. At first glance it works fine (something stays in cache thats why I didn't spot that earlier) however after checking this with keyboard I also noticed that on Android it stops to be focusable at all.

File: packages/block-editor/src/components/block-mobile-floating-toolbar/index.native.js
Comment (jbinda) [reply packages/block-editor/src/components/block-mobile-floating-toolbar/index.native.js]: moved

File: packages/block-editor/src/components/block-mobile-floating-toolbar/nav-up-icon.js
Comment (jbinda) [reply packages/block-editor/src/components/block-mobile-floating-toolbar/nav-up-icon.js]: I keep it according to our discussion before

File: packages/edit-post/src/components/layout/index.native.js
Comment (jbinda) [inline packages/edit-post/src/components/layout/index.native.js:157]: It seems moving `FloatingToolbar.Slot` here make it unfocusable at all on Android (see [this comment](https://github.com/WordPress/gutenberg/pull/21446#issuecomment-614136356))

**Comment by pinarol** [comment]
@iamthomasbishop  unfortunately we don’t have a customizable scroll mechanism on Android. Android just adjusts itself to the viewport, what I mean by viewport is this red frame:

Screen Shot 2020-04-16 at 12 29 23

I tried a simple mock to demonstrate how it’d look if we added the height of floating toolbar(assume it is the yellow view) to the equation:

Screen Shot 2020-04-16 at 13 16 07

you see how it just clips the viewport height?
so actually the whole mechanism just works by clipping the viewport, there’s no actual “extra scrolling” going on that we can control.

on iOS we do have that extra scrolling mechanism because we had to implement the whole autoscroll mechanism from scratch. We had no choice since on iOS we don’t have a system supported keyboard avoiding mechanism(at least on the components react-native allows us to use). it took months to get it working the way it works now.

on Android it is all system managed, we just clip the viewport and it automatically adjusts the scroll position according to where the caret is

So if we want to keep the FloatingToolbar at the bottom, we can consider dedicating the whole rectangle area to it by changing the design. (forget about the yellow view for a second, imagine the whole red frame is the floating toolbar, except that it is not floating any more :( )

Screen Shot 2020-04-16 at 13 17 44

**Comment by pinarol** [comment]
> @pinarol Ok, that all makes sense. I think at this point, our options are to either consider a new design that would take up the full-width in an area at top of the keyboard or to anchor the UI at the top of the viewport. Is it possible to keep iOS bottom-positioned and Android top (because of the restrictions you mentioned above)? I don't think this is ideal, but it isn't the worst option. It could also allow us to test out both variations and see which performs better.

@iamthomasbishop You mean as a permanent solution? I am all cool with having a few releases with such a difference to see which one we like. But, at the end of this phase I'd really prefer if we decide on a single UI. Having different UI per platform doubles our maintenance efforts. (BTW, our maintenance efforts our are already quite high(~50%) so the more we increase it the less time we'll have for new features)

Do you want us to generate new builds for each platform so you can test both options side by side? iOS bottom-positioned and Android top. Maybe one option will stand out quickly when you try.

**Comment by iamthomasbishop** [comment]
> You mean as a permanent solution?

No, a temporary one.

> Do you want us to generate new builds for each platform so you can test both options side by side? iOS bottom-positioned and Android top. Maybe one option will stand out quickly when you try.

Precisely. I'm pretty certain bottom-positioned is going to win out, but until we solve the positioning challenge, this is workable.

**Comment by jbinda** [comment]
@pinarol @iamthomasbishop 

I have update the FloatingToolbar according to discussion and now it works like below:

**Comment by pinarol** [comment]
> No, a temporary one.

 Cool 👌 

> but until we solve the positioning challenge

@iamthomasbishop Just for sake of clarity, does this mean until we come up with a new design that would take up the full-width in an area at top of the keyboard ?

**Comment by pinarol** [comment]
@jbinda Thank you for the updates. So, this is ready for another set of code reviews right?

**Comment by jbinda** [comment]
I will push updates in a minutes because I was dealing with the CI fails.

It's because how I import/export Breadcrumbs in FloatingToolbar. After refactor CI should pass and I will be able to generate bundles

File: packages/block-editor/src/components/block-list/index.native.js
Comment (jbinda) [inline packages/block-editor/src/components/block-list/index.native.js:146]: This will add additional space for FloatingToolbar when autoscroll to text base blocks

**Comment by jbinda** [comment]
@pinarol @iamthomasbishop 

I have generated production builds to present how FloatingToolbar works

You can download it here:
iOS: **waiting to finish build**
Andorid: [APK 45971](https://github.com/wordpress-mobile/WordPress-Android/pull/11692#issuecomment-615247791)

**Comment by pinarol** [comment]
@dratwas @cameronvoell ready for another pass.

**Comment by iamthomasbishop** [comment]
@jbinda this is feeling pretty solid overall, other than some minor scroll position issues (mentioned below). At this point, I think I'm comfortable shipping with the toolbar (temporarily) positioned differently between platforms (iOS bottom, Android top) until we have some time with each version and can make a decision on which to go with. We can address the challenges presented on Android if bottom-positioned wins out.

## Blockers

- There are some small inconsistencies with scrolling position when selecting blocks but I think those aren't specific to the toolbar. Android especially doesn't seem to handle scroll position quite as smoothly. If these are quick refinements and related to this work, let's solve now — otherwise don't consider a blocker.
- [iOS] One thing I think we probably should do, however, is add a small bit of additional spacing between the selected block and the toolbar when the block is nearby. By this, I mean adjust the y-position when a new block is selected so that there is at least ~8px spacing between the bottom-positioned floatingToolbar and the bottom of the selected block. ([Example](https://cloudup.com/c1_davZxDjH))
- Not necessarily a "blocker" per sé, but I would like to adjust the `background-color` of the floating toolbar to be the same color as the block toolbar. If my color picker in Figma is accurate, that looks like `#2e2e2e` — if there's a variable that represents that, let's use that to be consistent.

## Non-blockers

- Can we scroll the canvas to the associated block when the selected block part of the breadcrumbs is tapped?
- It'd be nice to apply some subtle transitions, if possible:
    - Upon entry: `300ms` transition using `easeOut`, which would fade in and translate the y-position ~4-8px upward
    - Upon entry: quick `50ms` transition using `easeIn`, which would fade out and translate the y-position ~4-8px downward
    - [Here's](https://cloudup.com/c5zqWU5j0J6) what the animation would look like together

## Other

- Maybe unrelated, but when a new block is inserted on to the canvas, the scroll position does not adjust ([example](https://cloudup.com/ciDKoEgtkIP)). This happens on both platforms.

---

Great work on this! While this wasn't the original "ideal" toolbar, it is certainly an improvement to what currently exists. Thank you!

---

> Just for sake of clarity, does this mean until we come up with a new design that would take up the full-width in an area at top of the keyboard ?

@pinarol No, I meant if we can solve in a way that allows us to place the toolbar on the bottom edge, like iOS. Although as I mentioned above, I'm comfortable with differing positions for now, because it's a real improvement over what we currently have.

**Comment by pinarol** [comment]
> @pinarol No, I meant if we can solve in a way that allows us to place the toolbar on the bottom edge, like iOS. Although as I mentioned above, I'm comfortable with differing positions for now, because it's a real improvement over what we currently have.

Actually the placement at the bottom isn’t the problem but covering the caret is. I don’t even know if there’s a way on Android to customize the scroll amount relative to caret to prevent the floating toolbar from covering the caret. I wouldn’t bet on it to happen soon considering the amount of work it took on iOS. But i am no Android expert so I’d like to get some opinions from @hypest @maxme. One thing i know for sure is, if we make the floating toolbar take up the full-width on top of the current toolbar then there will be no problem.

**Comment by iamthomasbishop** [comment]
> Actually the placement at the bottom isn’t the problem but covering the caret is.

Yea, I just mean the general positioning of the UI component (bottom vs. top) — if/when we were able to address that on Android — but it sounds like that's not likely. So I am okay with shipping Android top-aligned, then we can iterate if it causes problems. I don't think top-aligned is necessarily a terrible option, just less discoverable and less reachable than bottom-aligned.

> One thing i know for sure is, if we make the floating toolbar take up the full-width on top of the current toolbar then there will be no problem.

I would still prefer a top-aligned truly "floating" toolbar over a docked full-width toolbar, but if this performs poorly I do have some ideas for how a full-width toolbar would look.

**Comment by jbinda** [comment]
## Blockers

> - There are some small inconsistencies with scrolling position when selecting blocks but I think those aren't specific to the toolbar. Android especially doesn't seem to handle scroll position quite as smoothly. If these are quick refinements and related to this work, let's solve now — otherwise don't consider a blocker.


> - [iOS] One thing I think we probably should do, however, is add a small bit of additional spacing between the selected block and the toolbar when the block is nearby. By this, I mean adjust the y-position when a new block is selected so that there is at least ~8px spacing between the bottom-positioned floatingToolbar and the bottom of the selected block. ([Example](https://cloudup.com/c1_davZxDjH))

Fixed!



> - Not necessarily a "blocker" per sé, but I would like to adjust the `background-color` of the floating toolbar to be the same color as the block toolbar. If my color picker in Figma is accurate, that looks like `#2e2e2e` — if there's a variable that represents that, let's use that to be consistent.

`gray-500` was used. I switched to `$background-dark-elevated` (equivalent of `#2e2e2e`)
 
See below screen for light and dark mode



## Non-blockers

>- Can we scroll the canvas to the associated block when the selected block part of the breadcrumbs is tapped?

I have started to investigate how to implement that according to our convo on Slack but do not have worked version because of focusing on behaviour with text blocks.

>- It'd be nice to apply some subtle transitions, if possible:
    - Upon entry: `300ms` transition using `easeOut`, which would fade in and translate the y-position ~4-8px upward
    - Upon entry: quick `50ms` transition using `easeIn`, which would fade out and translate the y-position ~4-8px downward
    - [Here's](https://cloudup.com/c5zqWU5j0J6) what the animation would look like together

It's doable and I think it's not a big deal. Do we want to include that feature in this PR @pinarol ?

## Others

>Maybe unrelated, but when a new block is inserted on to the canvas, the scroll position does not adjust (example). This happens on both platforms.

I don't think it's related to FloatingToolbar itself. It's rather something with autoscroll

**Comment by pinarol** [comment]
> > Can we scroll the canvas to the associated block when the selected block part of the breadcrumbs is tapped?
> I have started to investigate how to implement that according to our convo on Slack but do not have worked version because of focusing on behaviour with text blocks.

This was tried as a part of [this PR](https://github.com/WordPress/gutenberg/pull/18791) in a slightly different way(the aim was to scroll to the newly created block but the mechanism needed was the same) and it was reverted because of multiple problems. This is not a straightforward thing to do, so I suggest let's not block this PR for that. Buttons block would really benefit from having this version of the FloatingToolbar because the current one pushes the button down and it looks jumpy. Same for Columns when stacked horizontally.

**Comment by pinarol** [comment]
> > It'd be nice to apply some subtle transitions, if possible:
> > - Upon entry: 300ms transition using easeOut, which would fade in and translate the y-position ~4-8px upward
> > - Upon entry: quick 50ms transition using easeIn, which would fade out and translate the y-position ~4-8px downward
> > - Here's what the animation would look like together
>
> It's doable and I think it's not a big deal. Do we want to include that feature in this PR @pinarol ?
> 

I suggest starting implementing this right after this PR considering that this was listed below non-blockers. I really prefer to have this PR merged asap to get rid of the jumpiness of individual Button blocks when stacked horizontally with other Buttons.

**Comment by jbinda** [comment]
>I suggest starting implementing this right after this PR considering that this was listed below non-blockers. I really prefer to have this PR merged asap to get rid of the jumpiness of individual Button blocks when stacked horizontally with other Buttons.

Ok so I will prepare builds after adjustments pointed as a blockers. After it gets Thomas approve as well as code review approve from @dratwas it can be merge

File: packages/block-editor/src/components/block-list/subdirectory-icon.js
Comment (dratwas) [inline packages/block-editor/src/components/block-list/subdirectory-icon.js:16]: I'm wondering if some actions are taken when we pass an empty string to the `transfrom` and if it wouldn't be better to pass `undefined` instead. I'm just thinking aloud :)

File: packages/block-editor/src/components/floating-toolbar/index.native.js
Comment (dratwas) [inline packages/block-editor/src/components/floating-toolbar/index.native.js:1]: It's not a blocker but I think this file should be moved instead of copied. In the current approach, we lose history.

File: packages/block-editor/src/components/floating-toolbar/index.native.js
Comment (dratwas) [inline packages/block-editor/src/components/floating-toolbar/index.native.js:4]: I believe it is added automatically and can be removed :)

File: packages/block-editor/src/components/floating-toolbar/index.native.js
Comment (dratwas) [inline packages/block-editor/src/components/floating-toolbar/index.native.js:57]: I think it should be `selectedClientId` since the `clientId` is a bit confusing

File: packages/block-editor/src/components/floating-toolbar/index.native.js
Comment (dratwas) [inline packages/block-editor/src/components/floating-toolbar/index.native.js:59]: This could be removed after renaming the `clientId`

File: packages/block-editor/src/components/floating-toolbar/index.native.js
Comment (dratwas) [inline packages/block-editor/src/components/floating-toolbar/index.native.js:70]: I think it would be better to add one check before all selectors since all of them depends on `selectedClientId`

Something like 
```js
if( !! selectedClientId ) {
  const rootBlockId = getBlockHierarchyRootClientId( selectedClientId );
  const parentId = getBlockRootClientId( selectedClientId );
  const hasRootInnerBlocks = !! getBlockCount( rootBlockId );
  return {
    selectedClientId,
    showFloatingToolbar,
    parentId,
  }
}
return { }
```

Comment (jbinda) [reply packages/block-editor/src/components/floating-toolbar/index.native.js:70]: how about that:

```jsx
		const selectedClientId = getSelectedBlockClientId();

		if( ! selectedClientId ) return;

		const rootBlockId = getBlockHierarchyRootClientId( selectedClientId );

		return {
			selectedClientId,
			showFloatingToolbar: !! getBlockCount( rootBlockId ),
			parentId: getBlockRootClientId( selectedClientId ),
		};
```

File: packages/block-editor/src/components/floating-toolbar/index.native.js
Comment (jbinda) [reply packages/block-editor/src/components/floating-toolbar/index.native.js]: removed

File: packages/block-editor/src/components/floating-toolbar/index.native.js
Comment (jbinda) [reply packages/block-editor/src/components/floating-toolbar/index.native.js]: changed with below proposal

File: packages/block-editor/src/components/floating-toolbar/index.native.js
Comment (jbinda) [reply packages/block-editor/src/components/floating-toolbar/index.native.js]: done

File: packages/block-editor/src/components/floating-toolbar/index.native.js
Comment (jbinda) [reply packages/block-editor/src/components/floating-toolbar/index.native.js]: WDYT ?

**Comment by jbinda** [comment]
Refreshed build are here:
iOS: [still building](https://github.com/wordpress-mobile/WordPress-iOS/pull/13927#issuecomment-615245960)
Andorid: [APK 46171](https://github.com/wordpress-mobile/WordPress-Android/pull/11692#issuecomment-615247791)

@iamthomasbishop it also includes merge Button so you can check how FloatingToolbar works with them as well

**Comment by iamthomasbishop** [comment]
@jbinda This is feeling pretty solid! The visual changes look good, and I don't think any of the additional requests I had should stop this from iteration shipping (the animations, scrolling to the selected block, etc). Great work!!

File: packages/block-editor/src/components/floating-toolbar/index.native.js
Comment (dratwas) [reply packages/block-editor/src/components/floating-toolbar/index.native.js]: LGTM :)

**Comment by iamthomasbishop** [comment]
@jbinda One tiny thing: I noticed while testing on Android that the toolbar background isn't using the updated `$background-dark-elevated` color. I tried un-installing and re-installing to make sure I wasn't on the previous build, but I'm on the updated one afaik.

**Comment by jbinda** [comment]
@iamthomasbishop ok I will check that and fix

**Comment by iamthomasbishop** [comment]
Oh, that's odd. Maybe I have the wrong build on Android 🤔 I am seeing this on Android:

| **Light** | **Dark** |
| --------------- | --------------- |
| <a href="https://user-images.githubusercontent.com/1200432/79870185-37ac3b00-83a8-11ea-954c-131ba9c5234f.png"></a> | <a href="https://user-images.githubusercontent.com/1200432/79870257-54487300-83a8-11ea-89d2-3368399982e3.png"></a> |

**Comment by jbinda** [comment]
As discussed on Slack I have also spot that issue on different Emulator with Android 9 (which does not support dar theme) and be able to fix that so probably will prepare fresh build (only for Android) to confirm on your side.

File: packages/edit-post/src/components/layout/index.native.js
Comment (cameronvoell) [inline packages/edit-post/src/components/layout/index.native.js:156]: Cleaner here in `layout` than in the old BlockListBlock location, nice!

**Review by cameronvoell — APPROVED** [review]
This change is working well in my tests on iOS and Android. Top of screen location on Android seems like a workable solution to me, seeing the limitations related to keyboard handling. Great work!

:shipit:

**Comment by jbinda** [comment]
Thank you al for checking this !

**Pull request merged**
